modules/store/src/action_with_modifiers_creator.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  ActionCreator,
+  ActionCreatorProps,
+  NotAllowedCheck,
+  TypedAction,
+} from './models';
+import { createAction } from './action_creator';
+
+export function createActionWithModifiers<
+  T extends string,
+  P extends object,
+  MetaProps extends any[]
+>(
+  type: T,
+  props: ActionCreatorProps<P> & NotAllowedCheck<P>,
+  ...modifiers: [
+    ...{
+      [i in keyof MetaProps]: (
+        p: P
+      ) => (
+        actionCreator: ActionCreator<T, (p: P) => TypedAction<T> & P>
+      ) => ActionCreator<T, (p: P) => TypedAction<T> & P & MetaProps[i]>;
+    }
+  ]
+) {
```

Comment:
Typescript infers this function as producing an action creator which returns an action that has properties of type `P` and the _intersection_ of all the properties of the payloads of the modifiers.

It would be desired to have the _union_ instead, but I tried using `MetaProps[number]` in a return signature only to continue to get the intersection.

Reply:
Essentially, if modifier A attaches "propertyOne" and modifier B attaches "propertyOne" and "propertyTwo", then Typescript will think the instantiated action can only have "propertyOne" available, not "propertyTwo."